src/operations/References.cpp

```diff
@@ -510,14 +351,37 @@ lsp::ReferenceResult WorkspaceFolder::references(const lsp::ReferenceParams& par
     {
         if (auto prefix = reference->prefix)
         {
-            if (auto importedModuleName = module->getModuleScope()->importedModules.find(prefix.value().value);
-                importedModuleName != module->getModuleScope()->importedModules.end())
+            auto requireInfo = findClosestAncestorModuleImport(*sourceModule, reference->prefix.value(), reference->prefixLocation->begin);
+            if (!requireInfo)
+                return std::nullopt;
+
+            auto [requireSymbol, requireExpr] = requireInfo.value();
+
+            if (reference->prefixLocation.value().containsClosed(position))
             {
-                auto references = findAllTypeReferences(importedModuleName->second, reference->name.value);
-                return processReferences(fileResolver, references);
+                // TODO: what if this symbol is returned! need to handle that so we can find cross-file references
+
+                auto references = findSymbolReferences(*sourceModule, requireSymbol);
+                result.reserve(references.size());
+                for (auto& location : references)
+                {
+                    result.emplace_back(lsp::Location{
+                        params.textDocument.uri, {textDocument->convertPosition(location.begin), textDocument->convertPosition(location.end)}});
+                }
+
+                return result;
             }
+            else
+            {
+                if (!reference->nameLocation.containsClosed(position))
+                    return std::nullopt;
 
-            return std::nullopt;
+                auto moduleInfo = frontend.moduleResolver.resolveModuleInfo(module->name, *requireExpr);
+                if (!moduleInfo)
+                    return std::nullopt;
```

Comment:
Lets use `modules->getModuleScope()->importedModules` to get this data like was here before, instead of calling out to the module resolver again. Then we also don't care about extracting `requireExpr`